### PR TITLE
Support private GKE clusters in gke deploy.sh script

### PR DESCRIPTION
**Pull request by ankushagarwal**
* Update gke/deploy.sh to support Private GKE Clusters using PRIVATE_CLUSTER env variable
* Minor changes to cluster-kubeflow.yaml

/cc @lluunn

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1192)

### CODE REVIEW

File: scripts/gke/deployment_manager_configs/cluster-kubeflow.yaml [hunk 1]
```diff
@@ -53,11 +53,9 @@ resources:
       # Whether to enable Pod Security Policy Admission Controller
       # Use v1beta1 api
       podSecurityPolicy: false
-      masterAuthorizedNetworksConfig:
-          cidrBlocks:
```
Comment (lluunn) [inline scripts/gke/deployment_manager_configs/cluster-kubeflow.yaml]: just curious: what's this?

Comment (ankushagarwal) [reply scripts/gke/deployment_manager_configs/cluster-kubeflow.yaml]: By default any IP is allowed to connect to the master. `masterAuthorizedNetworksConfig` is a fancy name for firewall rules for master.

**Pull request merged**